rest/route/permissions.go

```diff
@@ -0,0 +1,42 @@
+package route
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/gimlet"
+)
+
+type permissionsGetHandler struct{}
+
+func (p *permissionsGetHandler) Factory() gimlet.RouteHandler                     { return p }
+func (p *permissionsGetHandler) Parse(ctx context.Context, r *http.Request) error { return nil }
+
+func getPermissions(permissions []string) []model.APIPermission {
+	out := []model.APIPermission{}
+
+	for _, key := range permissions {
+		out = append(out, model.APIPermission{
+			Key:    key,
+			Name:   evergreen.GetDisplayNameForPermissionKey(key),
+			Levels: evergreen.GetPermissionLevelsForPermissionKey(key),
+		})
+	}
+
+	return out
+}
+
+func getAllPermissions() *model.APIPermissions {
+	out := &model.APIPermissions{}
+
+	out.ProjectPermissions = getPermissions(evergreen.ProjectPermissions)
+	out.DistroPermissions = getPermissions(evergreen.DistroPermissions)
```

Comment:
In order for this to work, if we ever add a new type of permissions (see other comment) we'd need to modify here, in addition to the evergreen permissions, and I think that is potentially unintuitive if you're editing the permissions in the globals file. I don't think there's anything to be done, but a comment in the evergreen pacakge would probably be good.

Reply:
I would also just put these function calls in the declaration of the out type, and maybe just do it all inline there.

Reply:
added a comment in evergreen